File: DemoApplication.Web/src/app/models/notifications/notification.model.ts
import { SeverityEnum, RolesEnum, EntityEnum } from "./../enums.model"

export class NotificationModel {
    constructor(
    public id?: number,
    public originator?: string,
    public originatorString?: string,
    public severity?: SeverityEnum,
    public description?: string,
    public date?: Date,
    public read?: boolean,
    public seenBy?: string,
    public seenByUser?: string,
    public seenAt?: Date,
    public forRole?: RolesEnum,
    public relatedId?: string,
    public relatedDescription?: string,
    public entity?: EntityEnum
    )   {
    }
}



// export interface Inotifications {
//   // interface to define a displayable notification
//     originator?: string,
//     severity?: string,
//     description?: string,
//     severityCode?: number,
//     date?: any
//   }

File: DemoApplication.Models/Notification/NotificationModel.cs
using System;
using DemoApplication.Enums;

namespace DemoApplication.Models;

/// <summary>
/// notification model for the database entity
/// </summary>
public class NotificationModel
{
    public Guid Id { get; set; }
    public Guid Originator { get; set; }
    public SeverityEnum Severity { get; set; }
    public string Description { get; set; }
    public DateTimeOffset Date { get; set; }
    public bool Read { get; set; }
    public Guid SeenBy { get; set; }
    public DateTimeOffset? SeenAt { get; set; }
    public RolesEnum ForRole { get; set; }
    public string RelatedId { get; set; }
    public EntityEnum Entity { get; set; }
}
